src/main/java/net/earthcomputer/clientcommands/command/MinesweeperCommand.java

```diff
@@ -0,0 +1,362 @@
+        public void tick() {
+            if (ticksPlaying > 0 && gameActive()) {
+                ticksPlaying += 1;
+            }
+        }
+
+        @Override
+        public boolean mouseClicked(double mouseX, double mouseY, int button) {
+            return false;
+        }
+
+        @Override
+        public boolean mouseReleased(double mouseX, double mouseY, int button) {
+
+            int tileX = Mth.floorDiv((int) (mouseX - topLeftX - 12), 16);
+            int tileY = Mth.floorDiv((int) (mouseY - topLeftY - 12), 16);
+
+            if (isWithinBounds(tileX, tileY) && gameActive()) {
+                if (button == InputConstants.MOUSE_BUTTON_LEFT) {
+                    if (ticksPlaying == 0) {
+                        generateMines(tileX, tileY);
+                        ticksPlaying = 1;
+                    }
+
+                    click(tileX, tileY);
+
+                    assert minecraft != null && minecraft.player != null;
+                    if (emptyTilesRemaining <= 0) {
+                        minecraft.player.playNotifySound(SoundEvents.NOTE_BLOCK_PLING.value(), SoundSource.MASTER, 1.0f, 2.0f);
+                    } else if (deathCoords != null) {
+                        minecraft.player.playNotifySound(SoundEvents.NOTE_BLOCK_BASS.value(), SoundSource.MASTER, 1.0f, 1.0f);
+                    }
+                } else if (button == InputConstants.MOUSE_BUTTON_RIGHT) {
+                    flag(tileX, tileY);
+                }
+            }
+
+            return true;
+        }
+
+        private boolean gameActive() {
+            return deathCoords == null && emptyTilesRemaining > 0;
+        }
+
+        private void generateMines(int avoidX, int avoidY) {
+            for (int i = 0; i < mines; i++) {
+                int x = RANDOM.nextInt(boardWidth);
+                int y = RANDOM.nextInt(boardHeight);
+
+                // too close to the clicked position
+                if (Mth.abs(avoidX - x) <= 1 && Mth.abs(avoidY - y) <= 1) {
+                    i--;
+                    continue;
+                }
+
+                // already a mine
+                if ((board[y * boardWidth + x] & 0b1100) >>> 2 == 2) {
+                    i--;
+                    continue;
+                }
+
+                incrementWarning(x - 1, y - 1);
+                incrementWarning(x, y - 1);
+                incrementWarning(x + 1, y - 1);
+
+                incrementWarning(x - 1, y);
+                board[y * boardWidth + x] = 0b000_10_0_0;
+                incrementWarning(x + 1, y);
+
+                incrementWarning(x - 1, y + 1);
+                incrementWarning(x, y + 1);
+                incrementWarning(x + 1, y + 1);
+            }
+        }
+
+        private void incrementWarning(int x, int y) {
+            if (isWithinBounds(x, y)) {
+                int idx = y * boardWidth + x;
+                byte original_tile = board[idx];
+                if ((original_tile & 0b1100) >>> 2 == 1) {
```

Comment:
I find these packed bit operations a bit confusing, especially without any documentation about the layout of the bit fields. Could you add some documentation on the `board` field, and add some helper methods to abstract these bit operations away?